### ISSUE TITLE
Allow users to turn off TLS verification

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/mail#2785]
### Feature Request

As of https://github.com/nextcloud/mail/pull/2782 TLS host verification is enabled by default. Admins can only disable it system-wide. It would be nice to have a user setting to disable it per account, so you can still have other accounts verified.

#### Summary

Make it possible to disable TLS verification on a user/account based instead of system-wide.

**Comment by beccon4** [comment]
>>If you want to return to the old insecure behavior, set app.mail.verify-tls-peer to true in config.php.<<
No - we have to set it to 0 (or false?) to get Mail to work with self signed certificates.

**Comment by ChristophWurst** [comment]
my bad. it should be set to `false` of course

**Comment by TeroKeso** [comment]
Is the same setting preventing to use "SMTP + security none" or what is preventing use SMTP no authentication settings?

![image](https://user-images.githubusercontent.com/29271631/78058770-77779800-7391-11ea-97b8-07dd56bb24c6.png)

**Comment by ChristophWurst** [comment]
Authentication mode *none* is not the same as no authentication. The app will still authenticate, but the connection won't be secured.

We currently only support authenticated connections.

**Comment by TeroKeso** [comment]
Thanks for clarifying that for me. Old versions did allow no authentication settings. Its better that it requires authentication, but old habits die slowly.

**Comment by ChristophWurst** [comment]
Are you sure? Which version?

**Comment by TeroKeso** [comment]
In version 15.0.x. Did get some errors, but the user could to read and change settings. We are not using this for production, just for educational purpose "Here is a good software for your own file hosting and communicational needs".  Lab network enforcing us to use all local solutions.  Its time to upgrade SMTP installation to include Lets Encrypt installation or other.

**Comment by ChristophWurst** [comment]
This app is independent of Nextcloud server. Which version of the app do you use?

**Comment by TeroKeso** [comment]
Now using (and having problems) with Nextcloud 18.0.3 and auto-installed version of Mail app. I am using Ubuntu 2004 LTS daily and its default install of PHP is preventing to use an older version. Some steps of the Nextcloud is assuming that user email box is working.

![image](https://user-images.githubusercontent.com/29271631/78117798-c44d8400-740e-11ea-99c4-fb4ccac77b01.png)

Our use case is marginal. Don't want to waste your time :) .Typically user have some access to public imps or SMTP servers. I Just don't want to waste time from teaching Nextcloud to doing more local postfix :)

**Comment by ChristophWurst** [comment]
Yeah, that also has nothing to do with the mail app. What you show is the email settings of Nextcloud for notifications. Consult help.nextcloud.com for support.

**Comment by TeroKeso** [comment]
I under that believe that that button works after you have activated admins mail app. If you have only one user (admin) you need to get admins mail app working before that button works. At least that is implied in the error message. 

Thanks for the help :)

**Comment by Cybnate** [comment]
Ensuring the PHP path for cert location is set correctly may also help to keep TLS on assuming you have a signed root certificate for your imap server:

php -r "print_r(openssl_get_cert_locations());"

By default this is different from the default location for root certificates in Ubuntu
